src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +526,201 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
     ]
 
     return client.execute_script_actions(resource_group_name, cluster_name, persist_on_success, script_actions_params)
+
+
+# pylint: disable=redefined-builtin
+def create_autoscale(cmd, client, resource_group_name, cluster_name, type, min_count=None,
+                     max_count=None, timezone=None, days=None, time=None, count=None):
+    from azure.mgmt.hdinsight.models import Autoscale, AutoscaleCapacity, AutoscaleRecurrence, AutoscaleSchedule, \
+        AutoscaleTimeAndCapacity
+    autoscale_configuration = None
+    if type == 'Load':
+        if not all([min_count, max_count]):
+            raise CLIError(
+                'When the autoscale type is Load, both the min_count and max_count should be specified.')
+
+        autoscale_configuration = Autoscale(
+            capacity=AutoscaleCapacity(
+                min_instance_count=min_count,
+                max_instance_count=max_count
+            )
+        )
+    elif type == 'Schedule':
+        if not all([timezone, days, time, count]):
+            raise CLIError(
+                'When the autoscale type is Schedule, all of the timezone, days, time, count should be specified.')
+
+        autoscale_configuration = Autoscale(
+            recurrence=AutoscaleRecurrence(
+                time_zone=timezone,
+                schedule=[AutoscaleSchedule(
+                    days=days,
+                    time_and_capacity=AutoscaleTimeAndCapacity(
+                        time=time,
+                        min_instance_count=count,
+                        max_instance_count=count
+                    )
+                )]
+            )
+        )
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale(cmd, client, resource_group_name, cluster_name, min_count=None, max_count=None, timezone=None):
+    from azure.mgmt.hdinsight.models import AutoscaleCapacity, AutoscaleRecurrence
+
+    cluster = client.get(resource_group_name, cluster_name)
+    autoscale_configuration = None
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError(
+            'Please use `az hdinsight autoscale create` to enable autoscale firstly.')
+
+    # try to update load-based configuration
+    if autoscale_configuration.capacity:
+        if min_count:
+            autoscale_configuration.capacity.min_instance_count = min_count
+        if max_count:
+            autoscale_configuration.capacity.max_instance_count = max_count
+
+    if not autoscale_configuration.capacity:
+        if min_count or max_count:
+            autoscale_configuration.capacity = AutoscaleCapacity()
```

Comment:
Do you want to update just max_count or min_count?
1. After using `AutoscaleCapacity()`, `autoscale_configuration.capacity` will be initialized, is right? 
2. Need we to compare new min/max value with original min/max vaule. 

If my understanding is wrong, just ignore it.

Reply:
For 2, we don't need to compare, because it is under `if not autoscale_configuration.capacity` which means there is not orgin value.

For 1, customer can specify --min-count or --max-count or both, so we need to initalize capacity firstly before we update.

---
In reply to: [466801729](https://github.com/Azure/azure-cli/pull/14692#discussion_r466801729) [](ancestors = 466801729)

Reply:
If you are confused with "not", it is a bug, I have fixed.

---
In reply to: [466803343](https://github.com/Azure/azure-cli/pull/14692#discussion_r466803343) [](ancestors = 466803343,466801729)

Reply:
Got it. Cool

---
In reply to: [466803343](https://github.com/Azure/azure-cli/pull/14692#discussion_r466803343) [](ancestors = 466803343,466801729)